src/Profiles.ts

```diff
@@ -292,53 +264,22 @@ export class Profiles { // Processing stops if there are no profiles detected
     }
 
     private async saveProfile(ProfileInfo, ProfileName, ProfileType) {
-        const mainZoweDir = path.join(require.resolve("@brightside/core"), "..", "..", "..", "..");
-        // we have to mock a few things to get the Imperative.init to work properly
-        try {
-            (process.mainModule as any).filename = require.resolve("@brightside/core");
-        } catch (error) {
-            vscode.window.showErrorMessage(error.message);
-        }
-        try {
-            ((process.mainModule as any).paths as any).unshift(mainZoweDir);
-        } catch (error) {
-            vscode.window.showErrorMessage(error.message);
-        }
-        if (this.initValue === -1) {
-            try {
-            // we need to call Imperative.init so that any installed credential manager plugins are loaded
-                await Imperative.init({ configurationModule: require.resolve("@brightside/core/lib/imperative.js") });
-            } catch (error) {
-                vscode.window.showErrorMessage(error.message);
-            }
-            this.initValue = 0;
-        }
         let zosmfProfile: IProfile;
         try {
-            zosmfProfile = await new CliProfileManager({
-                profileRootDirectory: path.join(ImperativeConfig.instance.cliHome, "profiles"),
-                type: "zosmf"
-            }).save({ profile: ProfileInfo, name: ProfileName, type: ProfileType });
+            zosmfProfile = await (await this.getCliProfileManager("zosmf")).save({ profile: ProfileInfo, name: ProfileName, type: ProfileType });
         } catch (error) {
             vscode.window.showErrorMessage(error.message);
         }
         return zosmfProfile.profile;
     }
 
-    private isSpawnReqd() {
-        if (this.spawnValue === -1) {
-            const homedir = os.homedir();
-            this.spawnValue = 0;
-            try {
-                const fileName = path.join(homedir, ".zowe", "settings", "imperative.json");
-                const settings = JSON.parse(fs.readFileSync(fileName).toString());
-                const value = settings.overrides.CredentialManager;
-                this.spawnValue = value !== false ? 0 : 1;
-            } catch (error) {
-                // default to spawn
-                this.spawnValue = 0;
-            }
+    private async getCliProfileManager( type: string ) {
+        if (!this.profileManager) {
+            this.profileManager = await new CliProfileManager({
+                profileRootDirectory: path.join(os.homedir(), ".zowe", "profiles"),
+                type
+            });
         }
-        return this.spawnValue;
+        return this.profileManager;
```

Comment:
I suggest making this a public getter for the profileManager.
That way you can avoid having to initialize it in multiple places (and guarantee a single instance of it)

Reply:
I didn't really want it accessed from outside of this instance of Profiles. It is only referenced in here and can only be initialized once per type.